README.md

```diff
@@ -7,108 +7,162 @@
                         )
                     ]
                 }
             ]
         },
         Production {
             lhs: Nonterminal(
-                "B"
+                "base"
             ),
             rhs: [
                 Expression {
                     terms: [
                         Terminal(
-                            "D"
+                            "A"
+                        )
+                    ]
+                },
+                Expression {
+                    terms: [
+                        Terminal(
+                            "C"
+                        )
+                    ]
+                },
+                Expression {
+                    terms: [
+                        Terminal(
+                            "G"
                         )
                     ]
                 },
                 Expression {
                     terms: [
                         Terminal(
-                            "E"
+                            "T"
                         )
                     ]
                 }
             ]
         }
     ]
 }
+
 ```
 
-## Example
+Once the `Grammar` object is populated, to generate a random sentence from it
+call the object's generate function. `grammar.generate()`. For the above grammar
+you could expect something like `TGGC` or `AG`. 
+
+If the generate function can't find a production for a nonterminal it tries
+to evaluate it will print the identifer as a nonterminal, i.e. `<identifier>`.
+
+The generate function will return an error if it detects an infinite loop caused
+by a production such as `<PATTERN> := <PATTERN>`.
+
+## Parse Example
 
 ```rust
 extern crate bnf;
+use bnf::Grammar;
 
 fn main() {
     let input =
-        "<postal-address> ::= <name-part> <street-address> <zip-part>;
+        "<postal-address> ::= <name-part> <street-address> <zip-part>
 
               <name-part> ::= <personal-part> <last-name> <opt-suffix-part> <EOL>
-                            | <personal-part> <name-part>;
+                            | <personal-part> <name-part>
+
+          <personal-part> ::= <initial> \".\" | <first-name>
 
-          <personal-part> ::= <initial> \".\" | <first-name>;
+         <street-address> ::= <house-num> <street-name> <opt-apt-num> <EOL>
 
-         <street-address> ::= <house-num> <street-name> <opt-apt-num> <EOL>;
+               <zip-part> ::= <town-name> \",\" <state-code> <ZIP-code> <EOL>
 
-               <zip-part> ::= <town-name> \",\" <state-code> <ZIP-code> <EOL>;
+        <opt-suffix-part> ::= \"Sr.\" | \"Jr.\" | <roman-numeral> | \"\"
+            <opt-apt-num> ::= <apt-num> | \"\";"
 
-        <opt-suffix-part> ::= \"Sr.\" | \"Jr.\" | <roman-numeral> | \"\";
-            <opt-apt-num> ::= <apt-num> | \"\";";
+    let grammar = Grammar::from_str(input);
+    match grammar {
+        Ok(g) => println!("{:#?}", g),
+        Err(e) => println!("Failed to make grammar from String: {:?}", e),
```

Comment:
Do bnf errors impl display? If not, they should. If they already do, then just no debug `:?` is necessary. Is the debug easier to read than the impl display?

Reply:
Ah no they shouldn't need debug, it's just a bad habit I picked up to default to always use `:?`